public/app/features/playlist/playlists_ctrl.ts

```diff
@@ -6,7 +6,11 @@ export class PlaylistsCtrl {
   navModel: any;
 
   /** @ngInject */
-  constructor(private $scope, private backendSrv, navModelSrv) {
+  constructor(private $scope, private backendSrv, navModelSrv, playlistSrv) {
+    if (playlistSrv.isPlaying) {
```

Comment:
Not sure about this one. 

So the problem is we want to stop the playlist as soon as a user leaves the playlist, but there's no specific playlist-component so we don't really know when a user leaves the playlist? Maybe we should listen to route-changes and if the new route is not a dashboard-route, run that if-statement.

https://github.com/grafana/grafana/blob/85751d9875802d08d663fab0f3b1e762befa938b/public/app/routes/GrafanaCtrl.ts#L148

Reply:
yes, think listenting to route changes is the way to go. And have short lives state flag when playlist srv changes dashboard to know when playlist is initiating a location change

Reply:
But we'd have a bug if a user starts a playlist and somehow click on a dashboard-link, then the playlist would go on.

Reply:
why would it?

Reply:
> And have short lives state flag when playlist srv changes dashboard to know when playlist is initiating a location change

Reply:
Didn't see your comment before I wrote mine. As long as the short lived state flag survives a full page reload it could work yea.

Reply:
@jschill why would it need to survive page reload? If you reload running playlist you will end up on a dashboard anyways.

Reply:
@dprokop Playlists do full page reloads now and then, but when i test your branch i see the reload is done against the playlist url (/playlist/play/<playlistId>) and not the current dashboard so the only thing that happens is that the playlist start over.

Reply:
https://github.com/grafana/grafana/blob/c2598e498a481b88f3361f4187e6ed397e3a9853/public/app/features/playlist/playlist_srv.ts#L33